qt/scientific_interfaces/ISISReflectometry/test/Reduction/ValidateRowTest.h

```diff
@@ -159,16 +163,16 @@ class ValidateRowTest : public CxxTest::TestSuite {
   }
 
   void testParseThetaFailsForNegativeAndZeroValues() {
-    TS_ASSERT_EQUALS(boost::none, parseTheta("-0.01"));
-    TS_ASSERT_EQUALS(boost::none, parseTheta("-0.12"));
-    TS_ASSERT_EQUALS(boost::none, parseTheta("-1"));
-    TS_ASSERT_EQUALS(boost::none, parseTheta("0.0"));
+    TS_ASSERT_EQUALS(std::nullopt, parseTheta("-0.01"));
+    TS_ASSERT_EQUALS(std::nullopt, parseTheta("-0.12"));
+    TS_ASSERT_EQUALS(std::nullopt, parseTheta("-1"));
+    TS_ASSERT_EQUALS(std::nullopt, parseTheta("0.0"));
   }
 
   void testParseScaleFactor() {
-    TS_ASSERT_EQUALS(boost::none, parseScaleFactor("ABSC"));
-    TS_ASSERT_EQUALS(boost::none, parseScaleFactor("").get());
-    TS_ASSERT_EQUALS(0.1, parseScaleFactor("0.1").get().get());
+    TS_ASSERT(!parseScaleFactor("ABSC").second);
+    TS_ASSERT(parseScaleFactor("").second);
```

Comment:
Is this line testing the same thing as before? Does it need this extra line?
```suggestion
    TS_ASSERT(parseScaleFactor("").second);
    TS_ASSERT_EQUALS(std::nullopt, parseScaleFactor("").first());
```

Reply:
You mean between testing parameters "ABSC" and "". In that case is yes, they are testing two different things. "ABSC" is a wrong scale factor, so the cell should report invalid (the pair returns false). In the other case "", the cell is in a valid state (it's ok to be empty) but there is no scale factor so it is empty.

Reply:
I mean that previously it was checking that "" returned a `boost::none` but now it's checking that the validity bool is true, but not that it's a `std::nullopt`.

Reply:
ah yes, before the outer nested boost::optional was wrapping a boost::none to communicate essentially the same, that the cell is in valid state but empty. Anyway, I can add the check for std::nullopt too.